core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -646,6 +674,246 @@ public Api getApi() {
 
     }
 
+    /**
+     * A version range for {@code Warning}s indicates which versions of a given plugin are affected
+     * by it.
+     *
+     * {@link #name}, {@link #firstVersion} and {@link #lastVersion} fields are only used for administrator notices.
+     *
+     * The {@link #pattern} is used to determine whether a given warning applies to the current installation.
+     *
+     * @since TODO
+     */
+    public static final class WarningVersionRange {
+        /**
+         * Human-readable English name for this version range, e.g. 'regular', 'LTS', '2.6 line'.
+         */
+        @Nullable
```

Comment:
Nullable does not force the static analyzer to really check the filed. CheckForNull would be better, especially for the public API